src/cluster_legacy.c

```diff
@@ -3488,17 +3571,22 @@ void clusterReadHandler(connection *conn) {
 
     while (1) { /* Read as long as there is data to read. */
         rcvbuflen = link->rcvbuf_len;
-        if (rcvbuflen < 8) {
-            /* First, obtain the first 8 bytes to get the full message
-             * length. */
-            readlen = 8 - rcvbuflen;
+        if (rcvbuflen < RCVBUF_MIN_READ_LEN) {
+            /* First, obtain the first 16 bytes to get the full message
+             * length and type. */
+            readlen = RCVBUF_MIN_READ_LEN - rcvbuflen;
         } else {
             /* Finally read the full message. */
             hdr = (clusterMsg *)link->rcvbuf;
-            if (rcvbuflen == 8) {
+            uint16_t type = ntohs(hdr->type);
+            int is_msg_light = (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT);
+            if (rcvbuflen == RCVBUF_MIN_READ_LEN) {
                 /* Perform some sanity check on the message signature
-                 * and length. */
-                if (memcmp(hdr->sig, "RCmb", 4) != 0 || ntohl(hdr->totlen) < CLUSTERMSG_MIN_LEN) {
+                 * and length.
+                 * The minimum length for PUBLISH and  PUBLISHSHARD will be clusterMsgLight_MIN_LEN
+                 * as we are using the `clusterMsgLight` hdr. */
+                if (memcmp(hdr->sig, "RCmb", 4) != 0 || (!is_msg_light && (ntohl(hdr->totlen) < CLUSTERMSG_MIN_LEN)) ||
+                    (is_msg_light && (ntohl(hdr->totlen) < CLUSTERMSG_LIGHT_MIN_LEN))) {
```

Comment:
I don't know if Clang format will allow but I would prefer each of these condition on separate line. 

* Header check.
* Size check for regular message.
* Size check for light message.

Reply:
It did not allow me, I had formatted it that way and the auto formatter changed it to this again